### PR TITLE
Feature/lubridate verbs

### PR DESCRIPTION
**Pull request by jstanai**
Functionality Added:
- SparkR `generics.R` and `functions.R` added. Commented out functionality that already exists in Master.
- Lubridate.R added with most basic functionality added for handling namespace conflicts. (Still much to do).

Items to Address:
- dplyr count() and n() are still included **need to remove/update**
- Shall we remove commented out rank functions in `generics.R` and `functions.R`?

### CODE REVIEW

**Review by danzafar — CHANGES_REQUESTED** [review]
OK I got a chance to look through this. It seems like there are a few large tasks that need to be done before we can pull these functions in.

- Remove the unneeded generics and functions. Many if not all of these are already commented out.
- Update the example docs for these. Note, we will need to keep `\dont_run{...}`. This will be a significant contribution.
- Include some type of smart test framework that covers edge cases or underlying call patterns. SparkR does not have any tests for these so I'm not 100% worried about it, but we should strive for something.
- It looks like the dplyr version of `coalesce` was removed. That should be put back. Coalesce has methods for both data frames and Columns. They do entirely different things. It's confusing. 
- I see there are namespace conflicts with `cov`, `endsWith`, `startsWith`, `lag`, `sd`, `var`, `window`, and `atan2`. `lag` is already and S3 generic so we should just leverage that. `sd` might work once we get `var` implemented. For the others we might use the trick Hadley suggested to get around these namespace conflicts. I can handle that once everything else is done.

File: R/dplyr.R [hunk 1]
```diff
@@ -220,16 +220,6 @@ mutate.spark_tbl <- function(.data, ...) {
...
-#' @export
-coalesce <- function(...) {
-  UseMethod("coalesce")
-}
-
-#' @importFrom dplyr coalesce
-coalesce.default <- function(...) {
-  dplyr::coalesce(...)
-}
-
```
Comment (danzafar) [inline R/dplyr.R]: Got to undo this

Comment (jstanai) [reply R/dplyr.R]: Yes will do, I didn't realize they were needed in both places.

**Comment by jstanai** [comment]
All great points, will resolve.

**Comment by jstanai** [comment]
@danzafar 

I have tried to implement "lag" using the alternative methods. However, I am running into issues with dfToCols. Core error appears to be the following:

```
Caused by: java.lang.UnsupportedOperationException: Cannot evaluate expression: lag(input[1, double, true], 1, null)
```

So something is happening with mutate, and I can't quite track down the problem. The column, from what I was able to test, is OK and correctly is calling the spark sql function (it returns a jc object properly). However, it doesn't seem to covert in the dataframe on collection. 

```
 spark_iris_2 = spark_iris %>% mutate(test = tidyspark::lag(Sepal_Width)) 
Java ref type org.apache.spark.sql.Column id 1236 
> spark_iris_2 = spark_iris %>% mutate(test = tidyspark::lag(Sepal_Width)) %>% collect()
Java ref type org.apache.spark.sql.Column id 1255 
20/05/23 15:17:18 ERROR RBackendHandler: dfToCols on org.apache.spark.sql.api.r.SQLUtils failed
```

**Comment by jstanai** [comment]
@danzafar 

What do you want to see in terms of updated documentation? Most of the `functions.R` functions are currently documented. Could you give an example or two of functions with insufficient documentation and what you would like to see and I'll implement?
